webapp/hooks/useNeedsApproval.ts

```diff
@@ -0,0 +1,30 @@
+import { useAllowance } from 'hooks/useAllowance'
+import { Address } from 'viem'
+import { useAccount } from 'wagmi'
+
+export const useNeedsApproval = function ({
+  address,
+  amount,
+  spender,
+}: {
+  address: string
+  amount: bigint
+  spender: Address
+}) {
+  const { address: owner } = useAccount()
+
+  const {
+    data: allowance = BigInt(0),
+    isLoading,
+    status: allowanceStatus,
+    queryKey: allowanceQueryKey,
+  } = useAllowance(address, {
+    args: { owner, spender },
+  })
+
+  return {
+    allowanceQueryKey,
+    isLoadingAllowance: isLoading,
+    needsApproval: amount > allowance && allowanceStatus === 'success',
```

Comment:
If `allowanceStatus` is not `success`, `needsApproval` will be `false`, which can be not the intended behavior. Perhaps the status should be exposed as `useAllowance` does, so the user decides what to do.

Reply:
Improved here [19bf37e](https://github.com/hemilabs/ui-monorepo/pull/1125/commits/19bf37ec5b69aaa0926bf57bc744bfde67f8510d) created issue #1126 

Note that I still think that it is ok to check for `success`, as this hook may be called for native tokens. The query will be disabled for those scenarios, so `allowanceStatus` won't be `success`, and `needsApproval` result in false, which is correct.

Usages of this hook then will still need to check for `loading` and `error`, and that's why these 2 are now exposed (well, `loading` was already exposed heh)

Reply:
For native tokens, this hook should just return `false`. No need do any actual query.

Reply:
The query is not running. See [useAllowance](https://github.com/hemilabs/ui-monorepo/blob/543ddc46c31939b82f58aa16e8b643faba3c1be1/webapp/hooks/useAllowance.ts#L30). But in order to check here that it did not run, I need to check for `allowanceStatus` (or to check something at least 😆 )